File: Web/ProcessingTools.Web/Scripts/Areas/Data/Models/Shared/Synonym.js
function Synonym(id, name, languageId, status) {
    var self = this;

    self.id = id;
    self.name = ko.observable(name).extend({ required: true, minLength: 3 });
    self.languageId = ko.observable(languageId).extend({ required: true });
    self.status = ko.observable(status);
}

File: src/ProcessingTools/Data/Geo/Geo.Data.Entity/Models/Synonym.cs
namespace ProcessingTools.Geo.Data.Entity.Models
{
    using System.ComponentModel.DataAnnotations;
    using ProcessingTools.Constants.Data.Geo;
    using ProcessingTools.Models.Contracts;

    public abstract class Synonym : SystemInformation, INameableIntegerIdentifiable, ISynonym, IDataModel
    {
        [Key]
        public int Id { get; set; }

        [Required(AllowEmptyStrings = false)]
        [MinLength(ValidationConstants.MinimalLengthOfGeoName)]
        [MaxLength(ValidationConstants.MaximalLengthOfGeoName)]
        public string Name { get; set; }

        public int? LanguageCode { get; set; }
    }
}